plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -46,9 +46,8 @@ type PluginFactoryArgs struct {
 	HardPodAffinitySymmetricWeight int32
 }
 
-// MetadataProducerFactory produces MetadataProducer from the given args.
-// TODO: Rename this to PriorityMetadataProducerFactory.
-type MetadataProducerFactory func(PluginFactoryArgs) algorithm.MetadataProducer
+// PriorityMetadataProducerFactory produces PriorityMetadataProducer from the given args.
+type PriorityMetadataProducerFactory func(PluginFactoryArgs) algorithm.MetadataProducer
```

Comment:
We should rename this only after we have a specific type for PriorityMetadata. We don't have that today. That's the first step before renaming this. I am specifically referring to this:
https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/algorithm/types.go#L49

Reply:
oh, got it, make a misunderstanding, i should hold this PR. I think we could use this after all priority functions change to map/reduce pattern.